ross/rotor_assembly.py

```diff
@@ -2512,6 +2512,43 @@ def run_rubbing(self, **kwargs):
         return defect
 
     def run_crack(self, **kwargs):
+        """Execute the crack defect and generates the crack object on the back-end.
+
+        Parameters
+        ----------
+        **kwargs: dictionary
+        
+            **kwargs receives:
+                dt : float
+                    Time step
+                tI : float
+                    Initial time
+                tF : float
+                    Final time
+                cd : float
+                    Crack depth
+                n_crack : float
+                    Element where the crack is located
+                speed : float
+                    Operational speed of the machine
+                massunb : array
+                    Array with the unbalance magnitude. The unit is kg.m.
+                phaseunb : array
+                    Array with the unbalance phase. The unit is rad.
+                crack_type : string
+                    String containing type of crack model chosed. The avaible types are: Mayes and Gasch.
+                print_progress : bool
+                    Set it True, to print the time iterations and the total time spent, by default False.
+        Examples
+        --------
+        >>> from ross.defects.crack import crack_example
+        >>> probe1 = (14, 0)
+        >>> probe2 = (22, 0)
+        >>> response = crack_example()
+        >>> results = response.run_time_response()
+        >>> fig = response.plot_dfft(probe=[probe1, probe2], range_freq=[0, 100], yaxis_type="log")
+        >>> # fig.show()
+        """
```

Comment:
It looks like `run_crack` is not used in the example.
Also, I recommend adding the `@check_units` decorator to the function, this way the user will be able to pass pint quantities to the function. The way it works is detailed in the `@check_units` docstring:

```
    """Wrapper to check and convert units to base_units.
    If we use the check_units decorator in a function the arguments are checked,
    and if they are in the dictionary, they are converted to the 'default' unit given
    in the dictionary.
    For example:
    >>> units = {
    ... "L": "meter",
    ... }
    >>> @check_units
    ... def foo(L=None):
    ...     print(L)
    ...
    If we call the function with the argument as a float:
    >>> foo(L=0.5)
    0.5
    If we call the function with a pint.Quantity object the value is automatically
    converted to the default:
    >>> foo(L=Q_(0.5, 'inches'))
    0.0127
    """
```
Arguments such as `unbalance_magnitude` are already available in the units dictionary (see [here](https://github.com/ross-rotordynamics/ross/blob/fde96c36fbb087551a2dd78cec25f315f0243c10/ross/units.py#L20-L41)), so I recommend that you just change the argument name from `massunb` to `unbalance_magnitude`.
Other arguments can be added to the units dictionary, but I recommend that you first look to see if you can use one argument name that is already there so we can keep the user interface more simple.

Reply:
Hi Raphael, do you think that we could open an issue to fix this '@check_units' and change the argument name from massunb to unbalance_magnitude  in all defect classes after we close this PR?

Reply:
Yes, no problem.